pypdf/_writer.py

```diff
@@ -2341,17 +2341,45 @@ def pageMode(self, mode: PagemodeType) -> None:  # deprecated
         deprecation_with_replacement("pageMode", "page_mode", "3.0.0")
         self.page_mode = mode
 
-    def add_annotation(self, page_number: int, annotation: Dict[str, Any]) -> None:
+    def add_annotation(
+        self,
+        page: Union[int, PageObject],
```

Comment:
:warning:  That is a breaking change . Please undo it.

Reply:
I see what you mean. I will keep page_number

Reply:
@pubpub-zz Small reminder :-)